salt/states/module.py

```diff
@@ -319,13 +319,10 @@ def _call_function(name, returner=None, **kwargs):
     arg_type, na_type, kw_type = [], {}, False
     for funcset in reversed(kwargs.get('func_args') or []):
         if not isinstance(funcset, dict):
-            kw_type = True
-        if kw_type:
-            if isinstance(funcset, dict):
-                arg_type += funcset.values()
-                na_type.update(funcset)
-            else:
-                arg_type.append(funcset)
+            arg_type.append(funcset)
+        elif isinstance(funcset, dict):
+            arg_type += funcset.values()
+            na_type.update(funcset)
         else:
```

Comment:
Also, this `else` should be removed. The `func_kw.update(funcset)` was only happening when `kw_type` was `False`, which only happens when `funcset` is a dict. Due to the initial `isinstance` performed above the differing code, we already will know that `funcset` is a dict if we simply change the 2nd `isinstance` to an `else`.